rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverRmiRuleProvider.java

```diff
@@ -85,9 +85,12 @@ private void extractMetadata(GraphRewrite event, JavaTypeReferenceModel typeRefe
 
             // Create the source report for the RMI Implementation.
             JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
-            for (JavaSourceFileModel source : javaClassService.getJavaSource(rmiServiceModel.getImplementationClass().getQualifiedName()))
-            {
-                source.setGenerateSourceReport(true);
+            
+            if (rmiServiceModel != null && rmiServiceModel.getImplementationClass() != null) {
```

Comment:
Technically the first rmiServiceModel null check is unnecessary here, as it is always initalized by the getOrCreate call above. It isn't harmful, though.